engine/src/main/java/com/google/android/fhir/search/SearchDsl.kt

```diff
@@ -126,6 +128,28 @@ data class Search(val type: ResourceType, var count: Int? = null, var from: Int?
     sort = parameter
     this.order = order
   }
+
+  /**
+   * Provides limited support for the reverse chaining on [Search]
+   * [https://www.hl7.org/fhir/search.html#has] Typical example would be `Search all Patient that
+   * have Condition - Diabetes` Condition.subject is set to a reference to the Patient when
+   * Condition object is created. So, the search code will look like:
+   * ```
+   *     FhirEngine.search<Patient> {
+   *        has<Condition>(Condition.SUBJECT) {
+   *          filter(Condition.CODE, Coding("http://snomed.info/sct", "44054006", "Diabetes"))
+   *        }
+   *     }
+   * ```
+   */
```

Comment:
```suggestion
  /**
   * Provides limited support for the reverse chaining on [Search]
   * [https://www.hl7.org/fhir/search.html#has]. For example: search all Patient that
   * have Condition - Diabetes. This search uses the subject field in the Condition resource. Code snippet:
   * ```
   *     FhirEngine.search<Patient> {
   *        has<Condition>(Condition.SUBJECT) {
   *          filter(Condition.CODE, Coding("http://snomed.info/sct", "44054006", "Diabetes"))
   *        }
   *     }
   * ```
   */
```